src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -0,0 +1,144 @@
+<?php
+namespace StoragelessSession\Http;
+
+use DateTime;
+use Zend\Stratigility\MiddlewareInterface;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Http\Message\ResponseInterface as Response;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Builder;
+use StoragelessSession\Session\Data;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\ValidationData;
+use Lcobucci\JWT\Signer\Key;
+
+final class SessionMiddleware implements MiddlewareInterface
```

Comment:
> Well, this way you lose the point of middlewares that is framework agnostic.

No, others can simply type-hint `callable` here.

 > Last try: if we really need an interface for BC purposes, why not provide the interface in the package itself?

No, the coupling is wanted here.

What you are looking for is very simple, and it's simply to push the interface into a client/server/dispatcher PSR ;-)

I'm not really worried about the coupling here either btw, as the package is mostly composed by abstract types.